core/shared/utils/bh_common.c

```diff
@@ -5,6 +5,65 @@
 
 #include "bh_common.h"
 
+
+static char *
+align_ptr(char *src, unsigned int b)
+{
+    uintptr_t v = (uintptr_t)src;
+    uintptr_t m = b - 1;
+    return (char *)((v + m) & ~m);
+}
+
+/*
+Memory copy, align with word
+*/
+int
+b_memcpy_aw(void *s1, unsigned int s1max, const void *s2, unsigned int n)
+{
+    char *dest = (char *)s1;
+    char *src = (char *)s2;
+
+    char *pa = align_ptr(src, 4);
+    char *pb = align_ptr((src + n), 4);
+
+    if (pa > src) {
+        pa -= 4;
+    }
+
+    for (unsigned int *p = (unsigned int *)pa; p < (unsigned int *)pb; p++) {
+        unsigned int buff = *(p);
```

Comment:
Don't define the variables in the for sentences, define them at the beginning of the function or `{ ... }`.